python/packages/nisar/workflows/stage_watermask.py

```diff
@@ -289,6 +291,20 @@ def translate_watermask(vrt_filename, outpath, x_min, x_max, y_min, y_max):
     input_x_min, xres, _, input_y_max, _, yres = ds.GetGeoTransform()
     length = ds.GetRasterBand(1).YSize
     width = ds.GetRasterBand(1).XSize
+
+    # Declare lambda function to snap min/max X and Y
+    # coordinates over the water grid
+    snap_coord = lambda val, snap, offset, round_func: round_func(  # noqa: E731
+        float(val - offset) / snap) * snap + offset
```

Comment:
No need to convert to `float` -- the "true division" operator (`/`) in Python3 always performs floating-point division.

<img width="606" height="637" alt="image" src="https://github.com/user-attachments/assets/8b86cfca-0781-4be8-ac2e-dadc46e1e484" />

```suggestion
    snap_coord = lambda val, snap, offset, round_func: round_func(  # noqa: E731
        (val - offset) / snap) * snap + offset
```